src/client/render.ts

```diff
@@ -2,8 +2,8 @@ import type {
 	Component,
 	Props
 } from './index.d';
-import type {createRoot} from 'react-dom/client';
-import type {flushSync, render as renderOrig} from 'react-dom';
+import {createRoot} from 'react-dom/client';
```

Comment:
Have you checked that this doesn't end up with a specific version of react-dom/client to be bundled with the client bundle? Because that could cause runtime error if the consumer app uses a different version of react.

I guess the client is part of the build system, so it should use the same react the app does.

Reply:
Look at the finished file, it should be small, not containing the huge react libs.